fastjsonschema/ref_resolver.py

```diff
@@ -0,0 +1,158 @@
+"""
+JSON Schema URI resolution scopes and dereferencing
+
+Code adapted from https://github.com/Julian/jsonschema
+"""
+import contextlib
+import re
+from urllib import parse as urlparse
+from urllib.parse import unquote
+from urllib.request import urlopen
+import json
+
+import requests
+
+from .exceptions import JsonSchemaException
+
+
+def resolve_path(schema, fragment):
+    """
+    Return definition from path.
+
+    Path is unescaped according https://tools.ietf.org/html/rfc6901
+
+    :argument schema: the referrant schema document
+    :argument str fragment: a URI fragment to resolve within it
+    :returns: the retrieved schema definition
+
+    """
+    fragment = fragment.lstrip('/')
+    parts = unquote(fragment).split('/') if fragment else []
+    for part in parts:
+        part = part.replace('~1', '/').replace('~0', '~')

```

Comment:
What is ~1 or ~0?

Reply:
They are escaped characters... see: https://tools.ietf.org/html/rfc6901
see: test ref.json ""escaped pointer ref"" test case.

P.S:. I will go now and check rest of comments and commit patch tomorrow